kmath-tensors/src/commonMain/kotlin/space/kscience/kmath/tensors/core/algebras/DoubleAnalyticTensorAlgebra.kt

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2018-2021 KMath contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package space.kscience.kmath.tensors.core.algebras
+
+import space.kscience.kmath.tensors.api.AnalyticTensorAlgebra
+import space.kscience.kmath.tensors.api.Tensor
+import space.kscience.kmath.tensors.core.DoubleTensor
+import space.kscience.kmath.tensors.core.tensor
+import kotlin.math.*
+
+public object DoubleAnalyticTensorAlgebra :
```

Comment:
What is the reason behind splitting the functionality in implementation? It makes sense in interfaces, but not in implementation.

Reply:
We implemented just the doubles, but in future you would have `Tensor<Long>` and `div` does not make sense then, as most of linear albegra operations (such as SVD etc.), or <Tensor<Complex<Double>>` and `sqrt` wouldn't make sense there (as many analytic functions as well actually). Strictly speaking we could merge `TensorPartialDivisionTensorAlgebra` with `LinearOpsTensorAlgebra`. Don't forget the mathematical terminology here though: an Algebra is a vector space with multiplication. A division Algebra is a vector space with division definied for non zero elements (in our case it's partial because we cannot divide by tensors with 0 entries as was discussed several times).

Reply:
I agree that there are some operations from `AnalyticTensorAlgebra` like `mean`  or `variance` that could be moved to `TensorPartialDivisionAlgebra`

Reply:
@altavir in that regard I invite to move things around as you wish and create a PR. I don't have any preferences.

Reply:
We can change things after merge, I guess since it is dev branch. I think it makes sense to do one implementation object for doubles and then add one for Int.

Reply:
Ok I will have a look at this